docs/storage/volume_migration.md

```diff
@@ -0,0 +1,219 @@
+      domain:
+        devices:
+          disks:
+          - disk:
+              bus: virtio
+            name: datavolumedisk1
+          interfaces:
+          - masquerade: {}
+            name: default
+        resources:
+          requests:
+            memory: 128Mi
+      terminationGracePeriodSeconds: 0
+      volumes:
+      - dataVolume:
+          name: src-dv
+        name: datavolumedisk1
+      networks:
+      - name: default
+        pod: {}
+```
+
+The datavolume `src-dv` is migrated to the pvc `dest-pvc`:
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: dest-pvc
+spec:
+  accessModes:
+  - ReadWriteMany
+  resources:
+    requests:
+      storage: 5Gi
+```
+
+by updating the VM:
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  labels:
+    kubevirt.io/vm: vm-dv
+  name: vm-dv
+spec:
+  updateVolumesStrategy: Migration
+  running: true
+  template:
+    metadata:
+      labels:
+        kubevirt.io/vm: vm-dv
+    spec:
+      domain:
+        devices:
+          disks:
+          - disk:
+              bus: virtio
+            name: datavolumedisk1
+          interfaces:
+          - masquerade: {}
+            name: default
+        resources:
+          requests:
+            memory: 128Mi
+      terminationGracePeriodSeconds: 0
+      volumes:
+      - persistentVolumeClaim:
+          claimName: dest-pvc
+        name: datavolumedisk1
+      networks:
+      - name: default
+        pod: {}
+```
+
+
+The destination volume may be of a different type or size than the source. It is possible to migrate from and to a block volume as well as a filesystem volume.
+The destination volume should be equal to or larger than the source volume. However, the additional difference in the size of the destination volume is not instantly visible within the VM and must be manually resized because the guest is unaware of the migration.
+
+The volume migration depends on the `VolumeMigration` feature gate. To enable the feature gate, add the following to the KubeVirt CR:
+```yaml
+apiVersion: kubevirt.io/v1
+kind: KubeVirt
+spec:
+  configuration:
+    developerConfiguration:
+      featureGates:
+        - VMLiveUpdateFeatures
+        - VolumesUpdateStrategy
+        - VolumeMigration
+  workloadUpdateStrategy:
+    workloadUpdateMethods:
+    - LiveMigrate
+```
+
+## Datavolume template
+
+Updating a datavolume that is referenced by a datavolume template requires special caution. The volumes section must include a reference to the name of the datavolume template. This means that the datavolume templates must either be entirely deleted or updated as well.
+
+Example of updating the datavolume for the original VM in the first example:
+```yaml
+apiVersion: cdi.kubevirt.io/v1beta1
+kind: DataVolume
+metadata:
+  name: dst-dv
+spec:
+  source:
+      blank: {}
+  pvc:
+    accessModes:
+      - ReadWriteOnce
+    resources:
+      requests:
+        storage: 2Gi
+    storageClassName: local
```

Comment:
So if in the original VM, DV also uses the `local` storage class, then in `dst-dv` we can use another storage class?

Reply:
Yes, the PVC specification can differ from volume mode, to access mode, to storage class or size

Reply:
Thanks. Would it be good to use a different sc name in this example?